Source/SwiftLintFramework/Rules/OpeningBraceRule.swift

```diff
@@ -25,16 +36,89 @@ public struct OpeningBraceRule: Rule {
             "func abc()\n\t{ }",
             "[].map(){ $0 }",
             "[].map( { } )"
+        ],
+        corrections: [
+            "struct Rule{}\n": "struct Rule {}\n",
+            "struct Rule\n{\n}\n": "struct Rule {\n}\n",
+            "struct Rule\n\n\t{\n}\n": "struct Rule {\n}\n",
+            "struct Parent {\n\tstruct Child\n\t{\n\t\tlet foo: Int\n\t}\n}\n":
+                "struct Parent {\n\tstruct Child {\n\t\tlet foo: Int\n\t}\n}\n",
+            "[].map(){ $0 }\n": "[].map() { $0 }\n",
+            "[].map( { } )\n": "[].map({ } )\n",
         ]
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let pattern = "((?:[^( ]|[\\t\\n\\f\\r (] )\\{)"
-        let excludingKinds = SyntaxKind.commentAndStringKinds()
-
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.violatingOpeningBraceRanges().map {
             StyleViolation(ruleDescription: self.dynamicType.description,
                 location: Location(file: file, offset: $0.location))
         }
     }
+
+    public func correctFile(file: File) -> [Correction] {
+        let violatingRanges = file.ruleEnabledViolatingRanges(
+            file.violatingOpeningBraceRanges(),
+            forRule: self
+        )
+        return writeToFile(file, violatingRanges: violatingRanges)
+    }
+
+    private func writeToFile(file: File, violatingRanges: [NSRange]) -> [Correction] {
+        var correctedContents = file.contents
+        var adjustedOffsets = [Int]()
+
+        for violatingRange in violatingRanges.reverse() {
+            let (contents, adjustedRange) =
+            correctContents(correctedContents, violatingRange: violatingRange)
+
+            correctedContents = contents
```

Comment:
`let (correctedContents, adjustedRange) = ...` didn't work because it is using `let` thus is different from previously declared `correctedContents`.
Also, `(correctedContents, let adjustedRange) = ...` didn't work too, so I decided to use temporary value here.